contracts/LendingPoolCore.vy

```diff
@@ -151,13 +146,34 @@ def activeForRewards(_lender: address) -> bool:
 ##### EXTERNAL METHODS - NON-VIEW #####
 
 @external
-def __init__(
-    _erc20TokenContract: address
-):
+def __init__(_erc20TokenContract: address):
     assert _erc20TokenContract != empty(address), "The address is the zero address"
-
     self.owner = msg.sender
     self.erc20TokenContract = _erc20TokenContract
+    self.migrationDone = False
+
+
+@external
+def migrate(_from: address):
+    assert not self.migrationDone, "migration already done"
+    assert msg.sender == self.owner, "msg.sender is not the owner"
+    assert _from != empty(address), "_address is the zero address"
+    assert _from.is_contract, "LPCore is not a contract"
+
+    self.activeLenders = ILendingPoolCore(_from).activeLenders()
+    self.fundsAvailable = ILendingPoolCore(_from).fundsAvailable()
+    self.fundsInvested = ILendingPoolCore(_from).fundsInvested()
+    self.totalFundsInvested = ILendingPoolCore(_from).totalFundsInvested()
+    self.totalRewards = ILendingPoolCore(_from).totalRewards()
+    self.totalSharesBasisPoints = ILendingPoolCore(_from).totalSharesBasisPoints()
+
+    lenders: DynArray[address, 2**50] = ILendingPoolCore(_from).lendersArray()
+    for lender in lenders:
+        self.lenders.append(lender)
+        self.knownLenders[lender] = True
+        self.funds[lender] = ILendingPoolCore(_from).funds(lender)
```

Comment:
If there would be more than ~300k lenders, I believe this would break the block gas limit. This isn't a problem for our current number of users but it's still important to keep in mind.